src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,66 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkLogin(user);
+        checkPassword(user);
+        checkAge(user);
+        addOriginUser(user);
+        return user;
+    }
+
+    private static void checkLogin(User user) {
+        String exceptionMessage = "User login length should be at least 6 characters.";
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login can't be null.");
+        }
+        if (user.getLogin().isEmpty()) {
+            throw new RegistrationException(exceptionMessage);
+        }
+        if (user.getLogin().isBlank()) {
+            throw new RegistrationException(exceptionMessage);
+        }
+        if (user.getLogin().length() < 6) {
+            throw new RegistrationException(exceptionMessage);
+        }
```

Comment:
there is no sense of using both `isEmpty` and `isBlank` if you need to just assert login length

I would suggest to unite `isBlank` and `lenght` conditions and use a message like `User login should have at least 6 non-whitespace characters`